### PR TITLE
[staging] PLT-372: Add Heracles client in Metastore for fetching users efficiently

### PR DESCRIPTION
**Pull request by sumandas0**
## Add Heracles client in Metastore for fetching users efficiently

>We are right now fetching users from Keycloak directly with pagination of 500, but this is causing issue right now and Keycloak isn't paginating correctly. So only way right now to fix Keycloak pagination issue is to increase pagination size to 1500, but this isn't at all scalable. So we will be now fetching users from Heracles with proper pagination this will help us scaling authorizer well.

## Type of change
- [ ] Bug fix (fixes an issue)
- [x] Enhancement
## JIRA Issue
- https://atlanhq.atlassian.net/browse/PLT-372

### CODE REVIEW

File: auth-agents-common/src/main/java/org/apache/atlas/plugin/util/KeycloakUserStore.java [hunk 1]
```diff
@@ -415,13 +415,13 @@ public RangerRole call() throws Exception {
                 //get all users for Roles
                 Thread usersFetcher = new Thread(() -> {
                     int start = 0;
-                    int size = 500;
+                    int size = 100;
```
Comment (checkaayush) [inline auth-agents-common/src/main/java/org/apache/atlas/plugin/util/KeycloakUserStore.java:417]: Why is the 100 hardcoded here?

File: client-auth/src/main/java/org/apache/atlas/auth/client/auth/AbstractAuthClient.java [hunk 2]
```diff
@@ -26,31 +28,32 @@
...
 
-abstract class AbstractKeycloakClient {
+public class AbstractAuthClient {
 
-    private final static Logger LOG = LoggerFactory.getLogger(AbstractKeycloakClient.class);
+    private final static Logger LOG = LoggerFactory.getLogger(AbstractAuthClient.class);
     private static final Map<Integer, AtlasErrorCode> ERROR_CODE_MAP = new HashMap<>();
 
-    private static final int DEFAULT_KEYCLOAK_RETRY = 3;
+    private static final int DEFAULT_RETRY = 3;
```
Comment (checkaayush) [inline client-auth/src/main/java/org/apache/atlas/auth/client/auth/AbstractAuthClient.java:36]: nit: Better name would be `DEFAULT_RETRY_COUNT`

File: client-auth/src/main/java/org/apache/atlas/auth/client/auth/AbstractAuthClient.java [hunk 3]
```diff
@@ -80,7 +87,7 @@ public AbstractKeycloakClient(KeycloakConfig keycloakConfig) {
         Timer.Sample timerSample = this.metricUtils.start(rawPath);
         okhttp3.Response response = chain.proceed(request);
         this.metricUtils.recordHttpTimer(timerSample, request.method(), rawPath, response.code(),
-                INTEGRATION, KEYCLOAK);
+                INTEGRATION, AUTH);
```
[the remark this answers is not in the document]
Comment (sumandas0) [reply client-auth/src/main/java/org/apache/atlas/auth/client/auth/AbstractAuthClient.java]: Done

**Review by checkaayush — APPROVED** [review]
LGTM

**Review by n5nk — APPROVED** [review]
LGTM @sumandas0

**Pull request merged**